src/lib/action/entry-transform.ts

```diff
@@ -1,31 +1,52 @@
 import { APIAction } from './action'
 import { OfflineAPI } from '../offline-api'
 import Entry from '../entities/entry'
+import * as _ from 'lodash'
 
 class EntryTransformAction extends APIAction {
   private contentTypeId: string
   private fromFields: string[]
-  private toFields: string[]
-  private transform: Function
+  // private toFields: string[]
+  private transformEntryForLocale: Function
+  private shouldPublish: boolean
 
-  constructor (contentTypeId: string, fromFields: string[], toFields: string[], transform: Function) {
+  // constructor (contentTypeId: string, fromFields: string[], toFields: string[], transformation: Function) {
+  constructor (contentTypeId: string, fromFields: string[], transformation: Function, shouldPublish: boolean = true) {
     super()
     this.contentTypeId = contentTypeId
     this.fromFields = fromFields
-    this.toFields = toFields
-    this.transform = transform
+    // this.toFields = toFields
+    this.transformEntryForLocale = transformation
+    this.shouldPublish = shouldPublish
   }
 
   async applyTo (api: OfflineAPI) {
     const entries: Entry[] = await api.getEntriesForContentType(this.contentTypeId)
+    const locales: string[] = await api.getLocalesForSpace()
     for (const entry of entries) {
-      const inputs = this.fromFields.map((fieldId) => entry.fields[fieldId])
-      const outputs = await this.transform(inputs)
-      for (let i = 0; i < this.toFields.length; i++) {
-        entry.fields[this.toFields[i]] = outputs[i]
+      const inputs = _.pick(entry.fields, this.fromFields)
+      for (const locale of locales) {
+        let outputsForCurrentLocale
+        try {
+          outputsForCurrentLocale = await this.transformEntryForLocale(inputs, locale)
```

Comment:
Missing code to handle `undefined` the same way we say in the interface spec (right now it will crash at `Object.keys`)